pkg/segment/structs/segstructs.go

```diff
@@ -682,46 +682,119 @@ func (qtype QueryType) String() string {
 	}
 }
 
-var unsupportedFuncs = []utils.AggregateFunctions{
-	utils.Estdc, utils.EstdcError, utils.ExactPerc, utils.Perc, utils.UpperPerc,
-	utils.Median, utils.Mode, utils.Stdev, utils.Stdevp, utils.Sumsq,
-	utils.Var, utils.Varp, utils.First, utils.Last, utils.Earliest,
-	utils.EarliestTime, utils.Latest, utils.LatestTime, utils.StatsRate,
+type UnsupportedFuncChecker interface {
+	IsUnsupported(funcName interface{}) bool
+}
+
+var unsupportedStatsFuncs = map[utils.AggregateFunctions]struct{}{
+	utils.Estdc:        {},
+	utils.EstdcError:   {},
+	utils.ExactPerc:    {},
+	utils.Perc:         {},
+	utils.UpperPerc:    {},
+	utils.Median:       {},
+	utils.Mode:         {},
+	utils.Stdev:        {},
+	utils.Stdevp:       {},
+	utils.Sumsq:        {},
+	utils.Var:          {},
+	utils.Varp:         {},
+	utils.First:        {},
+	utils.Last:         {},
+	utils.Earliest:     {},
+	utils.EarliestTime: {},
+	utils.Latest:       {},
+	utils.LatestTime:   {},
+	utils.StatsRate:    {},
+}
+
+var unsupportedEvalFuncs = map[string]struct{}{
+	"mvappend":         {},
+	"mvcount":          {},
+	"mvdedup":          {},
+	"mvfilter":         {},
+	"mvfind":           {},
+	"mvindex":          {},
+	"mvjoin":           {},
+	"mvmap":            {},
+	"mvrange":          {},
+	"mvsort":           {},
+	"mvzip":            {},
+	"mv_to_json_array": {},
+	"random":           {},
+	"floor":            {},
+	"pi":               {},
+	"ln":               {},
+	"log":              {},
+	"sigfig":           {},
+	"pow":              {},
+}
+
+type StatsFuncChecker struct{}
+
+func (c StatsFuncChecker) IsUnsupported(funcName interface{}) bool {
+	if name, ok := funcName.(utils.AggregateFunctions); ok {
+		_, found := unsupportedStatsFuncs[name]
+		return found
+	}
+	return false
 }
 
-func isUnsupported(funcName utils.AggregateFunctions) bool {
-	for _, f := range unsupportedFuncs {
-		if f == funcName {
-			return true
-		}
+type EvalFuncChecker struct{}
+
+func (c EvalFuncChecker) IsUnsupported(funcName interface{}) bool {
+	if name, ok := funcName.(string); ok {
+		_, found := unsupportedEvalFuncs[name]
+		return found
 	}
 	return false
 }
 
-func CheckUnsupportedFunctions(agg *QueryAggregators) error {
-	if agg == nil {
-		return nil
-	}
+func CheckUnsupportedFunctions(post *QueryAggregators) error {
+
+	var statsChecker UnsupportedFuncChecker = StatsFuncChecker{}
+	var evalChecker UnsupportedFuncChecker = EvalFuncChecker{}
```

Comment:
I think setting these up as interfaces is more trouble than it's worth, because the `IsUnsupported()` functions take different parameter types so then you have to pass an `interface{}` so you lose some compile-time type checking and have to do it at runtime. And I'm not seeing much benefit you get from this.

So you should make the `IsUnsupported()` functions take the parameter type they will actually get, and you can remove the `UnsupportedFuncChecker` interface. 

You don't even really need these new types you made for the checkers, but feel free to keep them if you like having them